packages/gatsby-plugin-fastify/src/utils/routes.ts

```diff
@@ -23,6 +23,6 @@ export function buildRedirectUrlFromParameters(path: string, data: { [s: string]
       throw new Error("Could not find url parameter " + lookupString + " in passed data object");
     }
 
-    return replacement;
+    return encodeURIComponent(replacement);
```

Comment:
Since this isn't a concern of this function let's move this out to https://github.com/gatsby-uc/plugins/blob/c109eb284913f00c05423804e3de888724848409/packages/gatsby-plugin-fastify/src/plugins/redirects.ts#L67 where the redirect is actually made.

Reply:
Ah yes, back at my computer now. That is actually where I put it initially and it was not working. However, I think using `encodeURI` here instead of `encodeURIComponent` will do the trick. I will test this now.

Reply:
What I'm wondering is if we need to check if it has a domain and use component if it doesn't.

Reply:
@moonmeister I think I've resolved this properly using `encodeURI` in `redirects.ts` so that it handles both cases. I've added an external domain redirect test as well as a [breaking example to the bug repro](https://github.com/tsdexter/bug-gatsby-plugin-fastify-invalid-chars/commit/714392dac751ea56c76d61940db754f3881df1e0). Resolved in https://github.com/gatsby-uc/plugins/pull/328/commits/990bcfaa18efb8fd436e4599b0c0a45b7cb92274